src/vaadin-date-time-picker.html

```diff
@@ -753,7 +776,14 @@
          * You can override the `checkValidity` method for custom validations.
          */
         checkValidity() {
-          return this.$.customField.checkValidity();
+          const invalidFields = this.$.customField.inputs
+            .filter(input => !(input.checkValidity).call(input)).length > 0;
+          const inputsEmpty = this.required && this.$.customField.inputs.filter(el => !el.value).length > 0;
+
+          if (invalidFields || inputsEmpty) {
+            return false;
+          }
+          return true;
```

Comment:
small nit: Here the word "field" and "input" both are used for the same thing in different variables. Also the variables if any (not all) field is invalid or empty. I'd prefer more consistent and clear names like `hasInvalidFields` and `hasEmptyFields` so the if condition is easier to understand correctly. Without the "has" prefix you could also understand it as "all" invalid/empty instead of "any".

Reply:
Updated.